src/instructlab/model/backends/backends.py

```diff
@@ -167,6 +173,12 @@ def is_model_gguf(model_path: pathlib.Path) -> bool:
     # Third Party
     from gguf.constants import GGUF_MAGIC
 
+    if not model_path.is_file():
```

Comment:
Similar story here to my comment above: https://www.freecodecamp.org/news/how-to-check-if-a-file-exists-in-python/

Reply:
@nathan-weinberg updated

Reply:
Same comment as above

Reply:
Same code snippet suggestion here
```python
if not isinstance(model_path, pathlib.Path):
    raise TypeError("'model_path' must be of type 'pathlib.Path')
```

Reply:
Then below, can do something similar to the safetensors func
```python
try:
    with open(model_path, "rb") as f:
        # Memory-map the file on the first 4 bytes (this is where the magic number is)
        mmapped_file = mmap.mmap(f.fileno(), length=4, access=mmap.ACCESS_READ)
        # Read the first 4 bytes
        first_four_bytes = mmapped_file.read(4)
        # Convert the first four bytes to an integer
        first_four_bytes_int = int(struct.unpack("<I", first_four_bytes)[0])
        # Close the memory-mapped file
        mmapped_file.close()
        return first_four_bytes_int == GGUF_MAGIC
    except (FileNotFoundError, PermissionError) as e:
        logger.debug("Failed to read file: %s", e)
        return False
```
@booxter let me know your thoughts here as well ^^